src/main/java/org/codehaus/mojo/aspectj/AbstractAjcCompiler.java

```diff
@@ -409,13 +403,39 @@
     @Parameter( defaultValue = "false" )
     protected boolean forceAjcCompile;
 
-    /**
-     * Sets the arguments to be passed to the compiler.<br>
-     * Example: &lt;compilerArgs&gt; &lt;arg&gt;-Xmaxerrs=1000&lt;/arg&gt; &lt;arg&gt;-Xlint&lt;/arg&gt; &lt;arg&gt;-J-Duser.language=en_us&lt;/arg&gt;
-     * &lt;/compilerArgs&gt;
+  /**
+   * Sets additional compiler arguments, e.g.
+   * <pre>{@code
+   * <compilerArgs>
+   *   <arg>-Xmaxerrs=1000</arg>
+   *   <arg>-Xlint</arg>
+   *   <arg>-J-Duser.language=en_us</arg> 
+   * </compilerArgs>
+   * }</pre>
+   * This option can be used in case you want to use AJC options not (yet) supported by this plugin.
+   * <p>
+   * <b>Caveat:</b> Be careful when using this option and select the additional compiler arguments wisely, because
+   * behaviour is undefined if you add arguments which have already been added by the plugin using regular parameters
+   * or their default values. The resulting compiler command line will in that case contain duplicate arguments, which
+   * might be illegal depending on the specific argument. Do not expect to be able to manually override existing
+   * arguments using this option or to replace whole argument lists.
+   *
+   * @since 1.13
+   */
+  @Parameter
+  protected List<String> additionalCompilerArgs = new ArrayList<>();
+
+  /**
+     * Activates compiler preview features (e.g. sealed classes in Java 16) when used with a suitable JDK version
+     *
+     * @since 1.13
      */
-    @Parameter
-    protected List<String> compilerArgs = new ArrayList<>();
+    // TODO:
+    //   Create tickets for at least Eclipse IDE and IntelliJ IDEA to recognise this switch and import it as a compiler
+    //   and possibly runtime setting. As for AJDT, maybe we have to implement it ourselves, but actually I found no
+    //   references to the Maven module there, so I guess the import is implemented somewhere else.
+    @Parameter( defaultValue = "false" )
+    protected boolean enablePreview;
```

Comment:
IMO we should not expose this option as a parameter. Artifacts created with this flag are useless with any other Java version. Most like a lot of people are not aware of this restriction. Hence if people need it, they should provide it as an additionalCompilerArgument.

Reply:
I disagree. Why would we want to patronise AspectJ developers, as if they were more stupid or less informed than any average Java developer who has this parameter at her disposal? As a separate parameter it is also easier to document, which I did. You probably read:

> Activates compiler preview features (e.g. sealed classes in Java 16) ***when used with a suitable JDK version***

If that is not enough, I can explain it similarly to how I did in the AspectJ 1.9.7 release notes:

https://github.com/kriegaex/org.aspectj/blob/9131c030bd77797b3593dee984ac75821055a44d/docs/dist/doc/README-197.html#L41-L48

If we could agree on making the caveat more explicit, I would be glad to improve the parameter's description. But I disagree with deliberately hiding this important compiler switch from our users. AspectJ supported JVM preview features before, and it is important that developers use them in order to gain experience and provide feedback to both OpenJDK and AspectJ.

Bottom line: Treat your fellow developers as adults, not as minors who need your protection.

Reply:
There has been one option that sounded acceptable for the maven-compiler-plugin: only allow it for SNAPSHOTs. But that proposal has been dropped, I can't remember the reason.
Not exposing this as a parameter should reduce the chance of mistakes. People that are aware of preview options can do so.
You defend the developer that wants easy access for using the preview option, whereas I defend the community that wants to consume proper artifacts. To me the impact of invalid artifacts is more important than ease of a parameter.

Reply:
Sorry to repeat myself, but again, this is not Maven Compiler Plugin. I am unaware of any policy here that we should mimic Maven Compiler Plugin in all respects. until two days ago, this plugin did not even have an option to accept custom compiler parameters. This has just been committed, and in my AspectJ Maven fork these parameters existed before the new catch-all was added to this plugin. Now that the option exists, it is agreat excuse to block all further API extensions by simply saying "we have an option to add custom compiler args". Then this plugin is no longer an abstraction of AJC, but simply a tool to pass through low level options. The user always has to check both, the compiler CLI documentation and the plugin documentation. I do not like that idea so much.

Reply:
I pushed https://github.com/mojohaus/aspectj-maven-plugin/pull/100/commits/2cb9d40878fb91ff01ebf4b13546c7568ec7c0e5, trying to address your concerns by means of documentation instead of removing the parameter. Please also read my commit comment in order to better understand my reasoning.

Reply:
My `-1`, too for this parameter. If such an artefact gets into mvn central, it will just not work, unless the consumer (other dev) has the very same java version with preview features enabled as well.

Also, you can already enable it by using `additionalCompilerArgument`.
It was never a problem for users (devs) to use `additionalCompilerArgument`, why would it be now?

Reply:
It was a problem. I added the parameter on request by someone who needed it.

The existing release of this plugin (not any fork) supports only Java 8 and the parameter was introduced much later. The generic compiler argument parameter was committed here only 3 weeks ago: https://github.com/mojohaus/aspectj-maven-plugin/commit/ebcda4badcaeeadce11fa57e5148dc2e7973dcf0. So please let us not pretend that it existed for a long time, shall we?

And how can you say it was never a problem before? This project was dormant for so long, who would have noticed? Since Javac, ECJ and AJC support the new parameter, there was no release of this plugin (and still there is none). But I did notice, because I am working on AspectJ and also was maintaining a fork of this plugin. AspectJ has tests for this compiler option, so it is only natural to also provide users here with the option. In case someone asks for support, we immediately see it in the POM and can resolve it quickly. I am just contributing this back.

> If such an artefact gets into mvn central, it will just not work, unless the consumer (other dev) has the very same java version with preview features enabled as well.

Every developer who provides an artifact on Central should know this. Besides, if he would use `additionalCompilerArgument`, the effect would be the same. So if he wants to use it, he will. Let us not play worldwide developer police but treat fellow developers as adults, like I said before. You are implying they are kids who need to be protected from their own stupidity or immaturity. Where does this sense of superiority on you guys' part come from?

And why is my clear and comprehensive documentation not superior to an security by obscurity? Is my Javadoc not clear enough?